mesh-design.md

```diff
@@ -0,0 +1,270 @@
+initially to prevent the user from specifying a mesh collider, but that could be done in other ways without
+complicating the API. See [Other Considerations](#consolidating-the-collider-apis-into-actor-patch).
+
+For example, if I wanted to create a box button, I might do something like this:
+
+```ts
+const context: MRE.Context;
+const assets: MRE.AssetContainer;
+
+// With a mesh asset and auto collider:
+const actor = MRE.Actor.CreateEmpty(context, {
+	actor: {
+		name: 'button',
+		appearance: {
+			meshId: assets.createBoxMesh('button', 0.25, 0.25, 0.1).id
+		},
+		// Note: This is proposed syntax. Today you'd have to call setCollider after creation.
+		// I.e. actor.setCollider('auto')
+		collider: { geometry: { shape: 'auto' } as MRE.AutoColliderGeometry }
+	}
+});
+
+// This would be equivalent to the following without `auto`:
+actor = MRE.Actor.CreateEmpty(context, {
+	actor: {
+		name: 'button',
+		appearance: {
+			meshId: assets.createBoxMesh('button', 0.25, 0.25, 0.1).id
+		}
+	}
+});
+actor.setCollider('box', false, { x: 0, y: 0, z: 0 }, { x: 0.25, y: 0.25, z: 0.1 });
+
+// Or this without mesh assets:
+actor = MRE.Actor.CreatePrimitive(context, {
+	definition: {
+		shape: 'box',
+		dimensions: { x: 0.25, y: 0.25, z: 0.1 }
+	},
+	addCollider: true,
+	actor: { name: 'button' }
+});
+```
+
+## Other Considerations
+
+### Consolidating the collider APIs into actor patch
+
+As previously noted, `CreatePrimitive` has one way to specify a collider, `CreateFromGltf` has another,
+and `setCollider` has yet another. This setup was done specifically to make it difficult to change actor
+colliders after actor creation. However, based on my research and with the exception of mesh colliders,
+creating a collider at patch time is no worse performance-wise than creating one at creation time. As such,
+the collider APIs should be consolidated to follow the same semantics as any other actor property.
+
+So instead of the `addCollider` flag:
+
+```ts
+MRE.Actor.CreatePrimitive(context, {
+	definition: {
+		shape: 'box',
+		dimensions: { x: 0.25, y: 0.25, z: 0.1 }
+	},
+	addCollider: true,
+	actor: { name: 'button' }
+});
+```
+
+The user would describe the desired collider in the patch:
+
+```ts
+MRE.Actor.CreateEmpty(context, {
+	actor: {
+		name: 'button',
+		appearance: {
+			meshId: assets.createBoxMesh('button', 0.25, 0.25, 0.1).id
+		},
+		collider: { geometry: { shape: 'auto' } as MRE.AutoColliderGeometry }
+	}
+});
+```
+
+This could also allow collider shape modification on the fly:
+
+```ts
+actor.collider.geometry = {
+	shape: 'sphere',
+	radius: 0.5
+} as MRE.SphereColliderGeometry;
+```
+
+I would also recommend removing `mesh` as a shape type, so it can only be specified implicitly
+via `auto`. Or else rename `mesh` to `unmanaged` so it's clear that it can't be used. Alternatively,
+we could go in the other direction and allow mesh colliders to be created at any time, and let
+the developer determine if it's appropriate to use.
```

Comment:
I think removing mesh in favor of auto is a better choice here and is what I would personally like to see.